plugin/pkg/auth/authenticator/token/tokenfile/tokenfile_test.go

```diff
@@ -139,3 +140,134 @@ func newWithContents(t *testing.T, contents string) (auth *TokenAuthenticator, e
 
 	return NewCSV(f.Name())
 }
+
+type inner func(*testing.T, *TokenAuthenticator)
+
+func testChangeTokenFileWrapper(t *testing.T, before string, after string, fn1 inner, fn2 inner) {
+	done := make(chan bool)
+	f, err := ioutil.TempFile("", "tokenfile_test")
+	if err != nil {
+		t.Fatalf("unexpected error creating tokenfile: %v", err)
+	}
+	f.Close()
+	defer os.Remove(f.Name())
+
+	go func() {
+		ioutil.WriteFile(f.Name(), []byte(before), 0644)
+		done <- true
+	}()
+	<-done
+	auth, err := NewCSV(f.Name())
+	fn1(t, auth)
+	time.Sleep(100 * time.Millisecond)
+	go func() {
+		ioutil.WriteFile(f.Name(), []byte(after), 0644)
+		done <- true
+	}()
+	<-done
+	time.Sleep(100 * time.Millisecond)
```

Comment:
during automated testing, on a busy test machine, this might not be long enough, which will make this test flaky.
Suggest making the test retry fn2 several times, until it passes, or a long time has passed.